src/Components/Users/UserAdd.tsx

```diff
@@ -247,21 +248,23 @@ export const UserAdd = (props: UserProps) => {
     dispatch({ type: "set_form", form });
   }
 
+  const handleMultiSelect = (event: any) => {
+    const { name, value } = event.target;
+    const form = { ...state.form };
+    form[name] = value;
+    dispatch({ type: "set_form", form });
+  }
+
   const validateForm = () => {
     let errors = { ...initError };
     let invalidForm = false;
     Object.keys(state.form).forEach(field => {
       switch (field) {
         case "facilities":
-          if (userType === "Staff" && state.form["user_type"] === "Staff") {
+
+          if (state.form[field].length === 0 && (userType === "Staff" || userType === "StaffReadOnly") && (state.form["user_type"] === "Staff" || state.form["user_type"] === "StaffReadOnly")) {
```

Comment:
Can we refactor this long `if` conditions?

Reply:
For now, this is the only use case that invalidates the form. How else can we change it? @smartclash

Reply:
Probably move that condition to a function and then add roles into an array while conditionally looping through the array? 

Can write up a suggestion tomorrow when I get back to my PC

Reply:
@smartclash any suggestions?